src/spikeinterface/sorters/launcher.py

```diff
@@ -146,12 +155,18 @@ def run_sorter_jobs(job_list, engine="loop", engine_kwargs={}, return_output=Fal
 
     elif engine == "slurm":
         # generate python script for slurm
-        tmp_script_folder = engine_kwargs["tmp_script_folder"]
+        tmp_script_folder = engine_kwargs.pop("tmp_script_folder")
```

Comment:
Here the popping is done so these arguments are not passed to sbatch. If using a subdict (comment above) I would revert these just to passing the variable so the code-reader does not have to track the state of `engine_kwargs` when reading.